src/elfcore.cc

```diff
@@ -747,16 +878,36 @@ static int CreateElfCore(void *handle, ssize_t (*writer)(void *, const void *, s
     }
     NO_INTR(sys_close(io.fd));
 
-    CountAUXV(&num_auxv, &vdso.address);
+    CountAUXV(&num_auxv, &vdso.address, main_pid, ext_process);
+    /*
+     * We check if we can read address pointed by vdso here.
+     */
+    if (ext_process && vdso.address) {
+      rc = PeekVDSO((void *)vdso.address, &vdso.ehdr, sizeof(Ehdr), main_pid);
+      if (rc == -1) {
+        vdso.ehdr.e_phnum = 0;
+#ifdef DEBUG
+        fprintf(stderr, "Could not peek to vdso, addr = %p err = %d\n", vdso.address, errno);
+#endif
+        vdso.address = 0;
+      }
+    }
     /* Read all mappings. This requires re-opening "/proc/self/maps"         */
     /* scope */ {
+      Phdr vdso_phdr[vdso.ehdr.e_phnum];
+      Phdr *vdso_phdr_ptr = &vdso_phdr[0];
       static const int PF_MASK = 0x00000007;
       struct {
         size_t start_address, end_address, offset, write_size;
         int flags;
       } mappings[num_mappings];
       io.data = io.end = 0;
-      NO_INTR(io.fd = sys_open("/proc/self/smaps", O_RDONLY, 0));
+      if (ext_process) {
```

Comment:
as per above, generally, this can be the same if you use pid instead of self.

Reply:
I am not comfortable using /proc/[pid]/ for self cases, as we are going to use the local address space of the process to dump the memory, which should match with the mappings that we are reading from the "maps" file. Otherwise we will have to use get_pid() and manipulate the path. Since we know the path for self is constant I don't want to spend cpu in path manipulation.